File: app/src/main/java/igrek/songbook/model/lyrics/LyricsLine.java
package igrek.songbook.model.lyrics;

import com.google.common.base.Joiner;

import java.util.ArrayList;
import java.util.List;

public class LyricsLine {
	
	private int y;
	
	private List<LyricsFragment> fragments = new ArrayList<>();
	
	public LyricsLine() {
	}
	
	public void addFragment(LyricsFragment fragment) {
		fragments.add(fragment);
	}
	
	public List<LyricsFragment> getFragments() {
		return fragments;
	}
	
	public int getY() {
		return y;
	}
	
	public void setY(int y) {
		this.y = y;
	}
	
	@Override
	public String toString() {
		return Joiner.on(" ").join(fragments);
	}
}

File: app/src/main/java/igrek/songbook/songpreview/lyrics/LyricsLine.kt
package igrek.songbook.songpreview.lyrics

import com.google.common.base.Joiner

data class LyricsLine(
        var y: Int = 0,
        var fragments: MutableList<LyricsFragment> = mutableListOf()
) {

    fun addFragment(fragment: LyricsFragment) {
        fragments.add(fragment)
    }

    override fun toString(): String {
        return Joiner.on(" ").join(fragments)
    }

    fun isBlank(): Boolean {
        return fragments.all { fragment -> fragment.text.isBlank() }
    }
}
